core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -2753,6 +2808,43 @@ private void resolveJoinColumns(QueryModel model) throws SqlException {
         }
     }
 
+    //Rewrite:
+    // sum(x*10) into sum(x) * 10, etc.
+    // sum(x+10) into sum(x) + count(x)*10
+    // sum(x-10) into sum(x) - count(x)*10
```

Comment:
This split may lead to an overflow that wouldn't be observed on the original function.

Reply:
Let me mark this one as nit. In any case, it sounds like a good idea to document these aggregate function transformations or at least document that they may be applied and the user should refer to the query plan.